### ISSUE TITLE
[BUG] Cannot create /dev/disk/azure/resource softlinks in Gen2 VM

### ISSUE THREAD

**Issue by yuxisun1217** [Azure/WALinuxAgent#1867]
**Describe the bug: A clear and concise description of what the bug is.**
In Gen2 VM, the 66-azure-storage.rules cannot create /dev/disk/azure/resource and /dev/disk/azure/root softlinks. 
**Gen1**:
```
# ll /dev/disk/azure/
lrwxrwxrwx. 1 root root  9 Apr 23 18:12 resource -> ../../sdb
lrwxrwxrwx. 1 root root 10 Apr 23 18:12 resource-part1 -> ../../sdb1
lrwxrwxrwx. 1 root root  9 Apr 23 18:12 root -> ../../sda
lrwxrwxrwx. 1 root root 10 Apr 23 18:12 root-part1 -> ../../sda1
lrwxrwxrwx. 1 root root 10 Apr 23 18:12 root-part2 -> ../../sda2
lrwxrwxrwx. 1 root root 10 Apr 23 18:12 root-part3 -> ../../sda3
lrwxrwxrwx. 1 root root 10 Apr 23 18:12 root-part4 -> ../../sda4
```
**Gen2**:
```
# ll /dev/disk/azure/
total 0
# ll /dev/disk/azure/scsi0/
total 0
lrwxrwxrwx. 1 root root 12 Apr 24 19:26 lun0 -> ../../../sda
lrwxrwxrwx. 1 root root 13 Apr 24 19:26 lun0-part1 -> ../../../sda1
lrwxrwxrwx. 1 root root 13 Apr 24 19:26 lun0-part2 -> ../../../sda2
lrwxrwxrwx. 1 root root 13 Apr 24 19:26 lun0-part3 -> ../../../sda3
lrwxrwxrwx. 1 root root 13 Apr 24 19:26 lun0-part4 -> ../../../sda4
lrwxrwxrwx. 1 root root 12 Apr 24 19:26 lun1 -> ../../../sdb
lrwxrwxrwx. 1 root root 13 Apr 24 19:26 lun1-part1 -> ../../../sdb1
drwxr-xr-x. 2 root root 180 Apr 24 19:26 scsi0
```
**Gen2 with cloud-init rule**:
```
# ll /dev/disk/cloud/
total 0
lrwxrwxrwx. 1 root root  9 Apr 24 18:54 azure_resource -> ../../sdb
lrwxrwxrwx. 1 root root 10 Apr 24 18:54 azure_resource-part1 -> ../../sdb1
lrwxrwxrwx. 1 root root  9 Apr 24 18:54 azure_root -> ../../sda
lrwxrwxrwx. 1 root root 10 Apr 24 18:54 azure_root-part1 -> ../../sda1
lrwxrwxrwx. 1 root root 10 Apr 24 18:54 azure_root-part2 -> ../../sda2
lrwxrwxrwx. 1 root root 10 Apr 24 18:54 azure_root-part3 -> ../../sda3
lrwxrwxrwx. 1 root root 10 Apr 24 18:54 azure_root-part4 -> ../../sda4
```

**Distro and WALinuxAgent details (please complete the following information):**
 - Distro and Version: RHEL-7/8
 - WALinuxAgent version: v2.2.46

**Comment by pgombar** [comment]
@trstringer could you take a look? Thanks!

**Comment by yuxisun1217** [comment]
Hi @trstringer ,
Thank you for your fix! I tested with the new rule in RHEL-8.3/7.9/gen1/gen2 VMs and all works well.

Gen1:
Old:
```
# ll /dev/disk/azure/
total 0
lrwxrwxrwx. 1 root root  9 Jul 21 15:28 resource -> ../../sdb
lrwxrwxrwx. 1 root root 10 Jul 21 15:28 resource-part1 -> ../../sdb1
lrwxrwxrwx. 1 root root  9 Jul 21 15:27 root -> ../../sda
lrwxrwxrwx. 1 root root 10 Jul 21 15:27 root-part1 -> ../../sda1
lrwxrwxrwx. 1 root root 10 Jul 21 15:27 root-part2 -> ../../sda2
lrwxrwxrwx. 1 root root 10 Jul 21 15:27 root-part3 -> ../../sda3
lrwxrwxrwx. 1 root root 10 Jul 21 15:27 root-part4 -> ../../sda4
```
New:
```
# ll /dev/disk/azure
total 0
lrwxrwxrwx. 1 root root  9 Jul 21 15:36 azure_resource -> ../../sdb
lrwxrwxrwx. 1 root root 10 Jul 21 15:36 azure_resource-part1 -> ../../sdb1
lrwxrwxrwx. 1 root root  9 Jul 21 15:35 azure_root -> ../../sda
lrwxrwxrwx. 1 root root 10 Jul 21 15:35 azure_root-part1 -> ../../sda1
lrwxrwxrwx. 1 root root 10 Jul 21 15:35 azure_root-part2 -> ../../sda2
lrwxrwxrwx. 1 root root 10 Jul 21 15:35 azure_root-part3 -> ../../sda3
lrwxrwxrwx. 1 root root 10 Jul 21 15:35 azure_root-part4 -> ../../sda4
```

Gen2:
Old:
```
# ll /dev/disk/azure/
total 0
drwxr-xr-x. 2 root root 180 Jul 21 15:30 scsi0
```
New:
```
# ll /dev/disk/azure/
total 0
lrwxrwxrwx. 1 root root  9 Jul 21 15:33 azure_resource -> ../../sdb
lrwxrwxrwx. 1 root root 10 Jul 21 15:33 azure_resource-part1 -> ../../sdb1
lrwxrwxrwx. 1 root root  9 Jul 21 15:33 azure_root -> ../../sda
lrwxrwxrwx. 1 root root 10 Jul 21 15:33 azure_root-part1 -> ../../sda1
lrwxrwxrwx. 1 root root 10 Jul 21 15:33 azure_root-part2 -> ../../sda2
lrwxrwxrwx. 1 root root 10 Jul 21 15:33 azure_root-part3 -> ../../sda3
lrwxrwxrwx. 1 root root 10 Jul 21 15:33 azure_root-part4 -> ../../sda4
```

**Comment by trstringer** [comment]
Great to hear! And thanks again for bringing this issue to our attention!
